### PR TITLE
Added aria-label to the ExpandMoreIcon to reflect dpl-react changes.

### PR DESCRIPTION
**Pull request by Dresse**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-637

#### Description

Added aria-label to the ExpandMoreIcon to reflect dpl-react changes.

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Additional comments or questions

This PR has a sister PR here: https://github.com/danskernesdigitalebibliotek/dpl-react/pull/1359

### CODE REVIEW

File: src/stories/Blocks/header/Header.tsx [hunk 1]
```diff
@@ -114,6 +114,7 @@ export const Header = (props: HeaderProps) => {
                   "header__menu-dropdown-icon--expanded": isDropdownOpen,
                 })}
                 onClick={() => setIsDropdownOpen(!isDropdownOpen)}
```
Comment (kasperbirch1) [inline src/stories/Blocks/header/Header.tsx:116]: The markup should be the same as in Drupal. Remove the onClick and wrap the element in a button ☺️

Comment (Dresse) [reply src/stories/Blocks/header/Header.tsx:118]: I have changed the markup to match Drupal. Button element now contains the onClick and is wrapping the icon.

File: src/stories/Blocks/header/Header.tsx [hunk 2]
```diff
@@ -101,20 +99,30 @@ export const Header = (props: HeaderProps) => {
           </nav>
           <div>
             <div className="header__menu-search">
-              <form className="header__menu-search-form">
```
Comment (kasperbirch1) [inline src/stories/Blocks/header/Header.tsx:104]: If `.header__menu-search-form` is no longer used in any part of the project, it should be deleted from `src/stories/Blocks/header/header.scss`

**Review by kasperbirch1 — APPROVED** [review]
I added a new comment about `.header__menu-search-form`, but otherwise, nice work! 👍

**Comment by Dresse** [comment]
> I added a new comment about `.header__menu-search-form`, but otherwise, nice work! 👍

Good thinking 😄 Have removed it now 👍

**Pull request merged**